### PR TITLE
Bootstrapper script for milo feds blocks

### PR DESCRIPTION
**Pull request by bandana147**
Adding a bootstrapper script which would initiate fed's global footer milo block which can be consumed by non milo client

JIRA - https://jira.corp.adobe.com/browse/MWPW-152861

Test:  [sample-page](https://bootstraper--federal--adobecom.hlx.page/local-dev/index.html?fedsBranch=bootstraper)

Integration flow would look like:

1. The consumer creates a folder under federal and creates their footer.docx file.
2. Adds contentRoot (e.g., /federal/foldername) and locale to window.fedsGlobalConfig.
3. Imports the script from https://bootstraper--federal--adobecom.hlx.page/federal/footer.js (this URL will change once merged on stage/main).

[Solution design](https://wiki.corp.adobe.com/pages/viewpage.action?pageId=3214883871#GlobalnavfromMiloasacomponent-3.SolutionDesign2:)

### CODE REVIEW

File: scripts/utils.js [hunk 1]
```diff
@@ -37,6 +57,29 @@ export const [setLibs, getLibs] = (() => {
...
+export async function bootstrapBlock(miloConfigs, blockConfig) {
+  if (!miloConfigs.url) {
+    console.log(`${blockConfig.label} url not found!`);
+    return;
+  }
+  const { miloLibs } = miloConfigs;
+  const { setConfig, createTag } = await import(`${miloLibs}/utils/utils.js`);
+  setConfig({ ...miloConfigs });
+  loadStyle(`${miloLibs}/blocks/${blockConfig.name}/${blockConfig.name}.css`);
+  loadStyle(`${miloLibs}/styles/styles.css`); //TODO: remove when css gets updated in core milo
```
Comment (sharmrj) [inline scripts/utils.js]: We might want to give it a callback to do some logging if either of these fail to load (to make it easier for us to diagnose if consumers have issues). The same probably goes for any other operations that might fail for any reason.

**Review by sharmrj — APPROVED** [review]
lgtm

**Comment by sharmrj** [comment]
I get a few of these issues in my console when I try to load the sample page:

"Access to fetch at 'https://www.stage.adobe.com/federal/footer/fragments/region-selector.plain.html' from origin 'https://bootstraper--federal--adobecom.hlx.page' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled."

Is this something we don't have to worry about on actual pages or do we need to set the 'Access-Control-Allow-Origin'  header (https://www.aem.live/docs/custom-headers)?

**Review by sonawanesnehal3 — APPROVED** [review]

**Review by salonijain3 — APPROVED** [review]

**Comment by bandana147** [comment]
> I get a few of these issues in my console when I try to load the sample page:
> 
> "Access to fetch at 'https://www.stage.adobe.com/federal/footer/fragments/region-selector.plain.html' from origin 'https://bootstraper--federal--adobecom.hlx.page' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled."
> 
> Is this something we don't have to worry about on actual pages or do we need to set the 'Access-Control-Allow-Origin' header (https://www.aem.live/docs/custom-headers)?

Yes, this wouldn't be an issue on actual page with whitelisted origins. But it should work for you with the sample page as well if you are connected to office vpn. @sharmrj

File: scripts/utils.js (hunk 1 above)
Comment (bandana147) [reply scripts/utils.js]: Yes, will add some loggers for all failure case.

**Review by nishantka — APPROVED** [review]

File: federal/footer.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const blockConfig = {
```
Comment (mokimo) [inline federal/footer.js:1]: This is an interesting case. The federal project was meant for strictly _content_ related use-cases.

I would therefor think this should live within the global navigation / footer code in milo, while the authoring should/could happen on the federal project such as `federal/globalnav/project-x/nav-1`

**Review by mokimo — COMMENTED** [review]
I have some concerns to have the code live in the federal repo, which is really "unregulated" compared to milo.

Code for this should live in milo core IMO as `adobecom/federal` project was just set up for technical reasons, so to enable the authoring portion on sharepoint.

On milo core, we provide better guardrails, a stage process, more thorough reviews etc. for code it's really a great home. It has longer ways, but prevents CSOs and should really be the place for our code, blocks & logic.

**Comment by salonijain3** [comment]
@mokimo Our understanding was that since this implementation of globalnav and footer is milo independent, it would be better if it is seperate and not regulated as part of Milo. This is closer to privacy standalone as compared to milo core. We can include the process / approvals / code reviews on this repo as well. 

cc: @narcis-radu

**Comment by narcis-radu** [comment]
Hey @salonijain3 - can you provide more details on why it would be better if the code is in a separate repository and not regulated as part of Milo? I see that in the current form we (would) need to duplicate code as well, which is not good on the long run.
If we're referring to privacy standalone, the only reason the project was moved outside of the main repository was that we could build it faster with no AEM dependencies and then consume the release version once everything was good. In the Milo world that's not a problem anymore though.
Federal has been set up as a centralised content solution and DAM, we are mixing up resources if we're introducing code that needs to be consumed.

**Comment by salonijain3** [comment]
@narcis-radu I was not aware that this repo was setup only as the content repo and I agree with the thought process. We can move it to milo. @sonawanesnehal3 and @bandana147 will be raising the PR shortly on milo.

**Pull request closed without merging**